File: Happypets - Spring/src/main/java/jwtspring/service/admin/AdminService.java
package jwtspring.service.admin;

import jwtspring.models.user.User;
import org.springframework.http.ResponseEntity;
import org.springframework.stereotype.Service;

import java.util.List;

@Service
public interface AdminService {

    ResponseEntity<List<User>> getUserList();

    ResponseEntity disableAccount(int userId);

    ResponseEntity enableAccount(int userId);
}

File: Happypets - Angular/src/app/services/admin/admin.service.ts
import {Injectable} from '@angular/core';
import {HttpClient} from '@angular/common/http';
import {Observable} from 'rxjs';

@Injectable({
  providedIn: 'root'
})
export class AdminService {
  const;
  API_URL = 'http://localhost:8080/api/admin/';

  constructor(private http: HttpClient) {
  }

  getUserList(): Observable<any> {
    return this.http.get(this.API_URL + 'user-list');
  }

  disableAccount(userId: any): Observable<any> {
    return this.http.post(this.API_URL + 'disable-account', userId);
  }

  enableAccount(userId: any): Observable<any> {
    return this.http.post(this.API_URL + 'enable-account', userId);

  }
}
